logger/src/lib.rs

```diff
@@ -22,20 +22,34 @@ impl log::Log for LoggerShim {
     fn flush(&self) {}
 }
 
-// Configures logging with a specific filter.
-// May be called at any time to re-configure the log filter
-pub fn setup_with_filter(filter: &str) {
-    let logger = env_logger::Builder::from_env(env_logger::Env::new().default_filter_or(filter))
-        .format_timestamp_nanos()
-        .build();
+fn replace_logger(logger: env_logger::Logger) {
     let max_level = logger.filter();
     log::set_max_level(max_level);
     let mut rw = LOGGER.write().unwrap();
     std::mem::replace(&mut *rw, logger);
     let _ = log::set_boxed_logger(Box::new(LoggerShim {}));
 }
 
-// Configures logging with the default filter ("error")
+// Configures logging with a specific filter overriding RUST_LOG.  _RUST_LOG is used instead
+// so if set it takes precedence.
+// May be called at any time to re-configure the log filter
+pub fn setup_with(filter: &str) {
+    let logger =
+        env_logger::Builder::from_env(env_logger::Env::new().filter_or("_RUST_LOG", filter))
```

Comment:
What's the use case for setting `_RUST_LOG`?

Reply:
I didn't see an api to set the log filter directly so I'm using a "commented out" environment variable.  it also serves a dual purpose where you can override RPC if need be.